src/compile-middleware.ts

```diff
@@ -83,36 +91,44 @@ export function babelCompile(forceCompile: boolean): RequestHandler {
       const compile = forceCompile || needCompilation(uaParser);
 
       if (compile) {
-        // TODO(justinfagnani): cache compilation results
+        const source = body;
+        const cached = babelCompileCache.get(source);
+        if (cached !== undefined) {
+          return cached;
+        }
         if (contentType === htmlMimeType) {
-          const document = parse5.parse(body);
-          const scriptTags = dom5.queryAll(document, isInlineJavaScript);
-          for (const scriptTag of scriptTags) {
-            try {
-              const script = dom5.getTextContent(scriptTag);
-              const compiledScriptResult = compileScript(script);
-              dom5.setTextContent(scriptTag, compiledScriptResult);
-            } catch (e) {
-              // By not setting textContent we keep the original script, which
-              // might work. We may want to fail the request so a better error
-              // shows up in the network panel of dev tools. If this is the main
-              // page we could also render a message in the browser.
-              console.warn(`Error compiling script in ${request.path}: ${e}`);
-            }
-          }
-          return parse5.serialize(document);
-        } else if (javaScriptMimeTypes.indexOf(contentType) >= 0) {
-          try {
-            return compileScript(body);
-          } catch (e) {
-            console.warn(`Error compiling script in ${request.path}: ${e}`);
-          }
+          body = compileHtml(source, request.path);
+        }
+        if (javaScriptMimeTypes.indexOf(contentType) !== -1) {
+          body = compileScript(source);
```

Comment:
Is there a reason we catch compile errors below but not here? Consider wrapping in try-catch